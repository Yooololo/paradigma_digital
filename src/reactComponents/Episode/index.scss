.podcast {
  &__episode {
    margin: $padding 0;
    width: calc(100% - 4 * $padding);
    border-radius: $box-border-radius;
    padding: $padding;
    box-shadow: $box-shadow;
    height: min-content;
  }

  .dark {
    h1 {
      color: $white;
    }
    div {
      color: $grey-light;
    }
  }
  .light {
    h1 {
      color: $black;
    }
    div {
      color: $grey-dark;
    }
  }
  h1 {
    font-size: 2 * $font-size;
    font-weight: 600;
    padding-bottom: $padding;
  }

  div {
    font-style: italic;
  }

  hr {
    margin: $padding 0;
  }

  audio {
    width: 100%;
    border-radius: 999px;
    background-color: $grey;
    &::-webkit-media-controls-enclosure {
      background-color: $grey;
    }
    &::-webkit-media-controls-panel,
    &::-webkit-media-controls-mute-button,
    &::-webkit-media-controls-play-button,
    &::-webkit-media-controls-timeline-container,
    &::-webkit-media-controls-current-time-display,
    &::-webkit-media-controls-time-remaining-display,
    &::-webkit-media-controls-timeline,
    &::-webkit-media-controls-volume-slider-container,
    &::-webkit-media-controls-volume-slider,
    &::-webkit-media-controls-seek-back-button,
    &::-webkit-media-controls-seek-forward-button,
    &::-webkit-media-controls-fullscreen-button,
    &::-webkit-media-controls-rewind-button,
    &::-webkit-media-controls-return-to-realtime-button,
    &::-webkit-media-controls-toggle-closed-captions-button {
      color: $white;
      font-size: $font-size;
      text-shadow: none;
    }
  }
}
