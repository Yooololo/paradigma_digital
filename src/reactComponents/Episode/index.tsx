import { useContext } from "react";
import { NavLink, useLocation } from "react-router-dom";
import parse from "html-react-parser";
import { EpisodeProps } from "../Podcast";
import { FormattedMessage } from "react-intl";
import { ThemeColorContext } from "../../context/ThemeColorWrapper";

const Episode = () => {
  const { themeColor } = useContext(ThemeColorContext);
  const {
    state: {
      episode,
      description: podcastDescription,
      artistName,
      podcastId,
      title,
    },
  }: {
    state: {
      episode: EpisodeProps;
      description: string;
      artistName: string;
      podcastId: string;
      title: string;
    };
  } = useLocation();

  return (
    <>
      <div className={`podcast ${themeColor}`}>
        <div className="podcast__detail">
          <NavLink
            state={{ id: podcastId, podcastDescription }}
            to={`/podcast/${podcastId}`}
          >
            <img src={episode.artworkUrl600} alt={artistName} />
          </NavLink>
          <hr />
          <section className="podcast__detail__section">
            <NavLink
              state={{ id: podcastId, podcastDescription }}
              to={`/podcast/${podcastId}`}
            >
              <p className={themeColor}>{episode.collectionName}</p>
              <p className={themeColor}>
                <FormattedMessage id="app.podcast.by" /> {artistName}
              </p>
            </NavLink>
          </section>
          <hr />
          <section className="podcast__detail__section">
            <p className={themeColor}>
              <FormattedMessage id="app.podcast.description" />
            </p>
            <p className={themeColor}>{podcastDescription}</p>
          </section>
        </div>
        <div className={`podcast__episode ${themeColor}`}>
          <h1>{title}</h1>
          <div>{parse(episode.description as string)}</div>
          <hr />
          <audio controls>
            <source
              src={episode.episodeUrl}
              type={`audio/${episode.episodeFileExtension}`}
            ></source>
          </audio>
        </div>
      </div>
    </>
  );
};

export default Episode;
