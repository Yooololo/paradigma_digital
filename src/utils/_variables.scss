// Colors
$grey-lighter: rgb(250, 250, 250);
$grey-light-50: rgb(225, 225, 225, 0.5);
$grey-light: rgb(200, 200, 200);
$grey: rgb(150, 150, 150);
$grey-dark: rgb(100, 100, 100);
$grey-darker: rgb(50, 50, 50);
$error-color: rgb(255, 0, 0);
$blue: rgb(25, 125, 175);
$blue-50: rgb(25, 125, 175, 0.5);
$black: #000000;
$white-50: rgba(255, 255, 255, 0.5);
$white: #ffffff;
$gold: #ffd700;

// Sizes
$podcast-box-height: 150px;
$podcast-detail-box-min-height: 500px;
$podcast-detail-box-max-width: 250px;
$box-margin: 100px;
$padding: 25px;
$margin: 10px;
$box-border-radius: 3px;

// Font
@import url("https://fonts.googleapis.com/css?family=Comfortaa:300,400,500,600,700&display=swap");
$font-family: "Comfortaa", sans-serif;
$font-family-alt: "Helvetica", "Arial", sans-serif;
$font-weight: 400;
$font-size: 16px;

// Utils
$box-shadow: 0 3px 4px 2px $grey, 0 -0.5px 0.5px 0.5px $grey-lighter;
$box-shadow-images: 0 2px 3px 1px $grey;
$animation: 0.5s;
