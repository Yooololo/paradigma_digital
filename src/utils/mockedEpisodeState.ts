export const mockedEpisodeState = {
  episode: {
    country: "USA",
    closedCaptioning: "none",
    artistIds: [121664449],
    artworkUrl600:
      "https://is5-ssl.mzstatic.com/image/thumb/Podcasts115/v4/0d/00/3d/0d003d69-71dc-54ac-2e49-71344d581ebc/mza_17414366298101550888.jpeg/600x600bb.jpg",
    previewUrl:
      "https://dts.podtrac.com/redirect.mp3/chrt.fm/track/8DB4DB/pdst.fm/e/nyt.simplecastaudio.com/9257fd86-22fe-4866-89a9-12c73a57a552/episodes/4fbc7178-cdf9-4956-baf7-ac9bfefa765d/audio/128/default.mp3?aid=rss_feed&awCollectionId=9257fd86-22fe-4866-89a9-12c73a57a552&awEpisodeId=4fbc7178-cdf9-4956-baf7-ac9bfefa765d&feed=W1rB_kgL",
    genres: [
      {
        name: "Music Commentary",
        id: "1523",
      },
    ],
    episodeGuid: "f4c2be9c-8e6d-4d41-940c-52ca05863771",
    description:
      "A conversation about the hip-hop eccentrics experiments, and the benefits (and perils) of collaborating outside an artists established genre. Guests: Jayson Buford and The Ringer's Justin Charity.",
    collectionId: 120315823,
    collectionName: "Popcast",
    trackId: 1000599098210,
    trackName: "Lil Yachty Breaks Boundaries and Takes on Rap Conservatism",
    shortDescription:
      "A conversation about the hip-hop eccentrics experiments, and the benefits (and perils) of collaborating outside an artists established genre. Guests: Jayson Buford and The Ringer's Justin Charity.",
    releaseDate: "2023-02-10T22:52:33Z",
    feedUrl: "https://feeds.simplecast.com/W1rB_kgL",
    kind: "podcast-episode",
    wrapperType: "podcastEpisode",
    artistViewUrl:
      "https://itunes.apple.com/us/artist/nytimes-com-podmaster/121664449?mt=2&uo=4",
    trackViewUrl:
      "https://podcasts.apple.com/us/podcast/lil-yachty-breaks-boundaries-and-takes-on-rap-conservatism/id120315823?i=1000599098210&uo=4",
    contentAdvisoryRating: "Clean",
    artworkUrl60:
      "https://is5-ssl.mzstatic.com/image/thumb/Podcasts115/v4/0d/00/3d/0d003d69-71dc-54ac-2e49-71344d581ebc/mza_17414366298101550888.jpeg/60x60bb.jpg",
    trackTimeMillis: 3264000,
    collectionViewUrl:
      "https://itunes.apple.com/us/podcast/popcast/id120315823?mt=2&uo=4",
    episodeUrl:
      "https://dts.podtrac.com/redirect.mp3/chrt.fm/track/8DB4DB/pdst.fm/e/nyt.simplecastaudio.com/9257fd86-22fe-4866-89a9-12c73a57a552/episodes/4fbc7178-cdf9-4956-baf7-ac9bfefa765d/audio/128/default.mp3?aid=rss_feed&awCollectionId=9257fd86-22fe-4866-89a9-12c73a57a552&awEpisodeId=4fbc7178-cdf9-4956-baf7-ac9bfefa765d&feed=W1rB_kgL",
    episodeFileExtension: "mp3",
    episodeContentType: "audio",
    artworkUrl160:
      "https://is5-ssl.mzstatic.com/image/thumb/Podcasts115/v4/0d/00/3d/0d003d69-71dc-54ac-2e49-71344d581ebc/mza_17414366298101550888.jpeg/160x160bb.jpg",
  },
  description:
    "The Popcast is hosted by Jon Caramanica, a pop music critic for The New York Times. It covers the latest in popular music criticism, trends and news.",
  artistName: "The New York Times",
  podcastId: "120315823",
  title: "Lil Yachty Breaks Boundaries and Takes on Rap Conservatism",
};
