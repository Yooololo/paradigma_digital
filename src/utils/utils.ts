import { EpisodeProps } from "../reactComponents/Podcast";

export const formatDate = (input: any) => {
  var date = input.match(/\d+/g),
    year = date[0].substring(2), // get only two digits
    month = date[1],
    day = date[2];

  return day + "/" + month + "/" + year;
};

export function format_msToHMS(input: number): string {
  const [MS_IN_SEC, SEC_IN_DAY, SEC_IN_HOUR, SEC_IN_MIN] = [
    1000, 86400, 3600, 60,
  ];
  let seconds = Math.round(Math.abs(input) / MS_IN_SEC);
  seconds = Math.floor(seconds % SEC_IN_DAY);
  const hours = Math.floor(seconds / SEC_IN_HOUR);
  seconds = Math.floor(seconds % SEC_IN_HOUR);
  const minutes = Math.floor(seconds / SEC_IN_MIN);
  seconds = Math.floor(seconds % SEC_IN_MIN);
  const [hh, mm, ss] = [hours, minutes, seconds].map((item) =>
    item < 10 ? "0" + item : item.toString()
  );
  return hh + ":" + mm + ":" + ss;
}

export const formatEpisodeData = (episode: EpisodeProps) => {
  let title = "";
  let date = "";
  let duration = "";
  let episodeId = "";
  if (episode.trackName) {
    title = episode.trackName;
  }
  if (episode.releaseDate) {
    date = episode?.releaseDate?.substring(0, 10);
    date = formatDate(date);
  }
  if (episode.trackTimeMillis) {
    duration = format_msToHMS(episode.trackTimeMillis);
  }
  if (episode.trackViewUrl) {
    episodeId = episode?.trackViewUrl?.split("i=")[1].split("&")[0];
  }
  return { title, date, duration, episodeId };
};
