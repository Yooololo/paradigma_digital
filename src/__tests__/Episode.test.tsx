import { mockedEpisodeState } from "../utils/mockedEpisodeState";
import { cleanup, render } from "@testing-library/react";
import Episode from "../reactComponents/Episode";
import { IntlProvider } from "react-intl";
import English from "../languages/en.json";
import { MemoryRouter } from "react-router-dom";

afterEach(cleanup);

describe("Episode", () => {
  it("should render the podcast episode and details", () => {
    const mockedState = mockedEpisodeState;

    const { container } = render(
      <MemoryRouter initialEntries={[{ state: mockedEpisodeState }]}>
        <IntlProvider locale="en-US" messages={English}>
          <Episode />
        </IntlProvider>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
    expect(mockedState.episode).toBeDefined();
    expect(mockedState.description).toBeDefined();
    expect(mockedState.artistName).toBeDefined();
    expect(mockedState.podcastId).toBeDefined();
    expect(mockedState.title).toBeDefined();
    expect(
      container.querySelectorAll(".podcast__detail__section")
    ).toHaveLength(2);
  });
});
