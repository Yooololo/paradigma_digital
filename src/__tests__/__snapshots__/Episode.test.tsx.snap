// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode should render the podcast episode and details 1`] = `
<div>
  <div
    class="podcast dark"
  >
    <div
      class="podcast__detail"
    >
      <a
        class=""
        href="/podcast/120315823"
      >
        <img
          src="https://is5-ssl.mzstatic.com/image/thumb/Podcasts115/v4/0d/00/3d/0d003d69-71dc-54ac-2e49-71344d581ebc/mza_17414366298101550888.jpeg/600x600bb.jpg"
        />
      </a>
      <hr />
      <section
        class="podcast__detail__section"
      >
        <a
          class=""
          href="/podcast/120315823"
        >
          <p
            class="dark"
          >
            Popcast
          </p>
          <p
            class="dark"
          >
            by
             
            The New York Times
          </p>
        </a>
      </section>
      <hr />
      <section
        class="podcast__detail__section"
      >
        <p
          class="dark"
        >
          Description:
        </p>
        <p
          class="dark"
        >
          The Popcast is hosted by Jon Caramanica, a pop music critic for The New York Times. It covers the latest in popular music criticism, trends and news.
        </p>
      </section>
    </div>
    <div
      class="podcast__episode dark"
    >
      <h1>
        Lil Yachty Breaks Boundaries and Takes on Rap Conservatism
      </h1>
      <div>
        A conversation about the hip-hop eccentrics experiments, and the benefits (and perils) of collaborating outside an artists established genre. Guests: Jayson Buford and The Ringer's Justin Charity.
      </div>
      <hr />
      <audio
        controls=""
      >
        <source
          src="https://dts.podtrac.com/redirect.mp3/chrt.fm/track/8DB4DB/pdst.fm/e/nyt.simplecastaudio.com/9257fd86-22fe-4866-89a9-12c73a57a552/episodes/4fbc7178-cdf9-4956-baf7-ac9bfefa765d/audio/128/default.mp3?aid=rss_feed&awCollectionId=9257fd86-22fe-4866-89a9-12c73a57a552&awEpisodeId=4fbc7178-cdf9-4956-baf7-ac9bfefa765d&feed=W1rB_kgL"
          type="audio/mp3"
        />
      </audio>
    </div>
  </div>
</div>
`;
